triple-lift.js

```diff
@@ -443,29 +598,117 @@ window.headertag.partnerScopes.push(function() {
-                    if (window.headertag.sizeRetargeting && window.headertag.sizeRetargeting[sizeKey]){
-                        width = window.headertag.sizeRetargeting[sizeKey][0];
-                        height = window.headertag.sizeRetargeting[sizeKey][1];
-                    }
-
-                    var ad = __creativeStore[id][width + 'x' + height].ad;
-
+                    var id = targetingMap[targetingKeys.idKey][0];
+                    var ad = creativeStore[id][width + 'x' + height].ad;
                     doc.write(ad);
                     doc.close();
                     if (doc.defaultView && doc.defaultView.frameElement) {
                         doc.defaultView.frameElement.width = width;
                         doc.defaultView.frameElement.height = height;
                     }
-                } catch (e){
+                } catch (e) {
                     //? if (DEBUG)
-                    console.log('Error trying to write ' + PARTNER_ID + ' ad to the page');
+                    console.log('Error trying to write ' + PARTNER_ID + ' ad to the page. Error: ' + e);
                 }
 
             }
         };
+
+
+        /* =============================================================================
+             * TL Helpers | Private methods
+             * -----------------------------------------------------------------------------
+             *
+             */
+
+
+        // TODO: Clean this up (break up, redo scope, consider timing)
+        function makeRequest(htSlotId, xSlot, numPlacements, callback) {
+            window.headertag.Network.ajax({
+                url: buildTplCall(xSlot),
+                method: 'GET',
+                partnerId: PARTNER_ID,
+                withCredentials: true,
+                onSuccess: function(response) {
+                    parseResponse(JSON.parse(response));
+                },
+                onFailure: function() {
+                    callback('TPL: demand request failed');
+                }
+            });
+
+            function parseResponse(response) {
+                // TODO validate response
+                var slotDemand = {}
+                if(!response.status) {
+                    var responseSize = response.width + 'x' + response.height;
+                    // store creative in creativeStore. blow away same response size for now
+                    creativeStore[htSlotId] = creativeStore[htSlotId] || {};
+                    creativeStore[htSlotId][responseSize] = {};
+                    creativeStore[htSlotId][responseSize].ad = response.ad;
+
+                    // format demand response to pass to callback
+                    slotDemand = {
+                        timestamp: Utils.now(),
+                        demand: {
+                            [targetingKeys.omKey]: responseSize + "_" + bidTransformer.transformBid(response.cpm),
+                            [targetingKeys.pmKey]: response.deal_id || null,
```

Comment:
okay, that generally makes sense. you should clarify in the docs how these keys are used / perhaps define them bit more robustly. Very easy to get lost in your terminology